pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go

```diff
@@ -46,5 +51,79 @@ func (r *ReconcileArtifactBuildRequest) Reconcile(ctx context.Context, request r
 	}
 	//TODO skeleton for now; start seeing what parts if any of build-request-processor or dependency-analyzer need to move here
 
+	// rough approximation of what is in https://github.com/redhat-appstudio/jvm-build-service/blob/main/build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/ProcessCommand.java
+	// where we replace the list done there, using the watch / relist induced event we get here
+	if abr.Status.State == v1alpha1.ArtifactBuildRequestStateNew {
+		abr.Status.State = v1alpha1.ArtifactBuildRequestStateMissing
+		err = r.client.Update(ctx, &abr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
+	//TODO need some golang approximation of RecipeRepositoryManager and RecipeGroupManager that takes
+	// the gav and abr and produces the needed result
+
+	dbName := "someNamDerivedFromArtifactBuildRequestAndGav"
+	dbNamespace := abr.Namespace
+	key := types.NamespacedName{Namespace: dbNamespace, Name: dbName}
+	db := v1alpha1.DependencyBuild{}
+	err = r.client.Get(ctx, key, &db)
+	if errors.IsNotFound(err) {
+		db.Spec.SCMURL = "someurl"
+		db.Spec.SCMType = "git"
+		db.Spec.Tag = "selectedTag"
+		db.Spec.Version = "someVersion"
+
+		err = r.client.Create(ctx, &db)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	abr.Status.State = v1alpha1.ArtifactBuildRequestStateBuilding
+	err = r.client.Update(ctx, &abr)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// see if we already launched a PipelineRun associated with this ABR
+	list := &pipelinev1beta1.PipelineRunList{}
+	abrNameForLabel := types.NamespacedName{Namespace: abr.Namespace, Name: abr.Name}.String()
+	lbls := map[string]string{"jvmbuildservice.hacbs.redhat.com/artifactbuildrequet": abrNameForLabel}
+	listOpts := &client.ListOptions{
+		Namespace:     abr.Namespace,
+		LabelSelector: labels.SelectorFromSet(lbls),
+	}
+	err = r.client.List(ctx, list, listOpts)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if len(list.Items) == 0 {
+		// create pipelinerun
+		pr := &pipelinev1beta1.PipelineRun{}
+		pr.Namespace = abr.Namespace
+		pr.GenerateName = abr.Name + "-"
+		//TODO fill in other needed fields of PR
+		err = r.client.Create(ctx, pr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		pr := list.Items[0]
+		if pr.Status.CompletionTime != nil {
+			abr.Status.State = v1alpha1.ArtifactBuildRequestStateComplete
+			condition := pr.Status.GetCondition(apis.ConditionSucceeded)
+			if !condition.IsTrue() {
+				abr.Status.State = v1alpha1.ArtifactBuildRequestStateFailed
+			}
+
+		}
```

Comment:
either a separate pipelinerun watcher, or triggering pipelinerun checks via intermediate artifactbuildrequest updates (which would be akin to what the shipwright buildrun reconciler does with associated taskrun updates) would follow in the completion time is not set